### PR TITLE
TSK-1486: Deleted unnecessary UnitTests of taskana-core

### PR DESCRIPTION
**Pull request by tge20**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [x] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [x] PR fulfills the ticket
- [x] Edge cases and unwanted side effects are tested
- [x] Readability

### CODE REVIEW

**Comment by tge20** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1486&id=tge20_taskana

**Comment by mustaphazorgati** [comment]
Let's walk through this together later

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java [hunk 1]
```diff
@@ -170,71 +168,6 @@ void testIdempotencyOfTaskCreation() throws Exception {
     assertThatThrownBy(call).isInstanceOf(TaskAlreadyExistException.class);
   }
 
-  @WithAccessId(user = "user-1-1")
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java:173]: why did you remove this?

File: lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java [hunk 2]
```diff
@@ -29,29 +28,24 @@
...
-  void testGetNoAttachmentSummaryFromTask() {
-    TaskService taskService = taskanaEngine.getTaskService();
-    List<TaskSummary> tasks = taskService.createTaskQuery().list();
-    assertThat(tasks).hasSize(30);
+    tasks = taskService.createTaskQuery().idIn("TKI:000000000000000000000000000000000066").list();
+    assertThat(tasks).hasSize(1);
 
-    List<AttachmentSummary> attachmentSummaries = tasks.get(0).getAttachmentSummaries();
+    attachmentSummaries = tasks.get(0).getAttachmentSummaries();
     assertThat(attachmentSummaries).isEmpty();
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java:43]: please make this a seperate test again.

File: lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java [hunk 3]
```diff
@@ -29,29 +28,24 @@
 
   @WithAccessId(user = "user-1-1")
   @Test
-  void testGetAttachmentSummariesFromTask() {
+  void should_ReturnTaskSummaryIncludingAttachmentsIfPresent_When_UsingTaskQuery() {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java:31]: please rework this name (after test separation)

File: lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java [hunk 4]
```diff
@@ -81,31 +75,4 @@ void should_ConfirmIfAttachmentSummariesAreCorrect_When_UsingTaskQueryAndGetTask
         .usingElementComparator(REFERENCE_COMPARATOR)
         .doesNotContainAnyElementsOf(originalAttachments); // but not same reference
   }
-
-  @WithAccessId(user = "user-1-1")
-  @Test
-  void should_ConfirmIfAttachmentSummariesAreCorrect() throws Exception {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java:87]: why did you delete this?

File: lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java [hunk 5]
```diff
@@ -81,31 +75,4 @@ void should_ConfirmIfAttachmentSummariesAreCorrect_When_UsingTaskQueryAndGetTask
...
-    TaskService taskService = taskanaEngine.getTaskService();
-    // find Task with ID TKI:00...00
-    List<TaskSummary> tasks =
-        taskService
-            .createTaskQuery()
-            .classificationKeyIn("T2000")
-            .customAttributeIn(TaskCustomField.CUSTOM_1, "custom1")
-            .list();
-    assertThat(tasks).hasSize(1);
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachmentAccTest.java:96]: `taskService.createTaskQuery().idIn("TKI:00...00").list()`

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java [hunk 6]
```diff
@@ -333,47 +327,6 @@ void testQueryForCustom7WithException() throws Exception {
     assertThat(result2).hasSize(2);
   }
 
-  @WithAccessId(user = "admin")
-  @Test
-  void testQueryTaskByCustomAttributes() throws Exception {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java:338]: Why did you remove this?

File: lib/taskana-core/src/test/java/pro/taskana/sampledata/SampleDataGeneratorTest.java [hunk 7]
```diff
@@ -1,26 +0,0 @@
-package pro.taskana.sampledata;
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/pro/taskana/sampledata/SampleDataGeneratorTest.java:1]: Please move this test to `taskana-common-data`

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/WorkbasketServiceImpl.java [hunk 8]
```diff
@@ -1044,8 +1044,8 @@ public void deleteWorkbasketAccessItemsForAccessId(String accessId)
    * @param workbasketImplToUpdate the workbasket to update
    * @throws ConcurrencyException if the workbasket has been modified by some other process.
    */
-  void checkModifiedHasNotChanged(Workbasket oldWorkbasket, WorkbasketImpl workbasketImplToUpdate)
-      throws ConcurrencyException {
+  public void checkModifiedHasNotChanged(
+      Workbasket oldWorkbasket, WorkbasketImpl workbasketImplToUpdate) throws ConcurrencyException {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/WorkbasketServiceImpl.java:1048]: why is this public?

File: lib/taskana-core/src/test/java/pro/taskana/classification/internal/models/ClassificationModelsCloneTest.java [hunk 9]
```diff
@@ -1,81 +0,0 @@
-package pro.taskana.classification.internal.models;
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/pro/taskana/classification/internal/models/ClassificationModelsCloneTest.java:1]: This test is important.

File: lib/taskana-core/src/test/java/pro/taskana/monitor/internal/ReportTest.java [hunk 10]
```diff
@@ -1,308 +0,0 @@
...
-import pro.taskana.monitor.api.reports.Report;
-import pro.taskana.monitor.api.reports.header.TimeIntervalColumnHeader;
-import pro.taskana.monitor.api.reports.item.MonitorQueryItem;
-import pro.taskana.monitor.api.reports.item.QueryItemPreprocessor;
-import pro.taskana.monitor.api.reports.row.FoldableRow;
-import pro.taskana.monitor.api.reports.row.Row;
-import pro.taskana.monitor.api.reports.row.SingleRow;
-
-/** Tests for {@link Report}. */
-class ReportTest {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/pro/taskana/monitor/internal/ReportTest.java:22]: This is important.

File: lib/taskana-core/src/test/java/pro/taskana/monitor/internal/WorkingDaysToDaysReportConverterTest.java [hunk 11]
```diff
@@ -1,334 +0,0 @@
...
-import java.util.List;
-import org.junit.jupiter.api.Test;
-
-import pro.taskana.common.api.CustomHoliday;
-import pro.taskana.common.api.WorkingDaysToDaysConverter;
-import pro.taskana.monitor.api.reports.header.TimeIntervalColumnHeader;
-import pro.taskana.monitor.internal.preprocessor.WorkingDaysToDaysReportConverter;
-
-/** Test for the DaysToWorkingDaysReportConverter. */
-class WorkingDaysToDaysReportConverterTest {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/pro/taskana/monitor/internal/WorkingDaysToDaysReportConverterTest.java:16]: This is important

File: lib/taskana-core/src/test/java/pro/taskana/task/internal/TaskAttachmentTest.java [hunk 12]
```diff
@@ -1,137 +0,0 @@
...
-import pro.taskana.task.api.models.AttachmentSummary;
-import pro.taskana.task.api.models.ObjectReference;
-import pro.taskana.task.internal.models.TaskImpl;
-
-/**
- * Unit Test for methods needed fot attachment at TaskImpl.<br>
- * This test should test every interaction with Attachments, which means adding, removing, nulling
- * them.
- */
-class TaskAttachmentTest {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/pro/taskana/task/internal/TaskAttachmentTest.java:22]: this is important.

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

**Review by mustaphazorgati — DISMISSED** [review]

**Review by mustaphazorgati — DISMISSED** [review]

**Review by mustaphazorgati — APPROVED** [review]

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**